src/averageAges.js

```diff
@@ -37,7 +41,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const personWithChildren = (withChildren)
+    ? people.filter(person => people
+      .some(child => child.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  return personWithChildren
```

Comment:
you can make universal function for counting average age for everyone

Reply:
+++